Cabal/Distribution/Package.hs

```diff
@@ -317,19 +353,24 @@ mkLegacyUnitId = newSimpleUnitId . mkComponentId . display
 data Dependency = Dependency PackageName VersionRange
                   deriving (Generic, Read, Show, Eq, Typeable, Data)
 
-instance Binary Dependency
+-- | Describes a legacy `build-tools`-style dependency on an executable
```

Comment:
It is "legacy" because we do not know if the build-tool referred to, refers to a pkg-config executable, or an internal executable (thus it is stringly typed.)